### PR TITLE
[OPEN DEV] Add Swiftlint to `BraintreeCore`

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

- Add `swiftlint.yml` to GitHub workflows
- Add `.swiftlint.yml` file with ruleset - this ruleset is the same as PPCP
- Add build script for `swiftlint`
- Address all swiftlint violations in `BraintreeCore` - will enable 1 module at a time to make PRs easier to reason about, additionally this allow us to progressively enable Swiftlint across the SDK while ensure code changes in modules where it's enable are compliant

### Checklist

- ~[ ] Added a changelog entry~

### Authors

- @jaxdesmarais

### CODE REVIEW

File: Braintree.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3154,6 +3155,26 @@
...
+			inputPaths = (
+			);
+			name = Swiftlint;
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "#!/bin/sh\n\nif test -d \"/opt/homebrew/bin/\"; then\n  PATH=\"/opt/homebrew/bin/:${PATH}\"\nfi\n\nexport PATH\n\nif which swiftlint >/dev/null; then\n  swiftlint\nelse\n  echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint\"\nfi\n";
```
Comment (scannillo) [inline Braintree.xcodeproj/project.pbxproj:3175]: I think we'll need to add instructions for downloading SwiftLint to these: our [Demo app running instructions](https://github.com/braintree/braintree_ios?tab=readme-ov-file#demo) & [test running instructions](https://github.com/braintree/braintree_ios/blob/main/DEVELOPMENT.md#tests)

Comment (jaxdesmarais) [reply Braintree.xcodeproj/project.pbxproj:3175]: Added: https://github.com/braintree/braintree_ios/pull/1364/commits/a5c84c58c7d81b2e408c21a214f93f034715af48

File: README.md [hunk 2]
```diff
@@ -84,6 +84,9 @@ Versions 4.9.6 and below use outdated SSL certificates and are unsupported.
 
 ## Demo
 
+### Prerequisites
```
Comment (scannillo) [inline README.md:87]: Nit - this makes it read that the actual steps to run the demo are in "Prerequisities". Maybe install swiftlint can just be step 1 before pod install (similar comment in the other guide)

**Review by scannillo — APPROVED** [review]
A few nits, otherwise this is very exciting!

File: .swiftlint.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+# Paths to include in lint
+included:
+  - Sources/BraintreeCore
+
+excluded:
+  - Sources/BraintreeCore/BTAPIPinnedCertificates.swift
+
+disabled_rules:
+  - todo
+  - type_name # tests will have have the format <SUT>_Tests
```
Comment (agedd) [inline .swiftlint.yml:14]: small nit: 
```suggestion
  - type_name # tests will have the format <SUT>_Tests
```

**Review by warmkesselj — APPROVED** [review]

**Review by agedd — APPROVED** [review]
this is so cool! 🚀

File: .swiftlint.yml [hunk 4]
```diff
@@ -0,0 +1,106 @@
+# Reference: https://github.com/realm/SwiftLint
+# Required Swiftlint Version
+# swiftlint_version: 0.39.2
```
Comment (jaxdesmarais) [inline .swiftlint.yml:3]:
```suggestion
# swiftlint_version: 0.55.1
```

**Pull request merged**